llvm/test/CodeGen/SPIRV/hlsl-intrinsics/clamp-vec.ll

```diff
@@ -0,0 +1,130 @@
+; RUN: llc  -verify-machineinstrs -O0 -mtriple=spirv-unknown-unknown %s -o - | FileCheck %s
+; RUN: %if spirv-tools %{ llc -O0 -mtriple=spirv-unknown-unknown %s -o - -filetype=obj | spirv-val %}
```

Comment:
What is the reason for having a clamp and clamp-vec test file? Why not merge to one file? We have done scalar and vector tests for all other hlsl intrinsics in one file so not sure what makes this special.

Reply:
I'm just following what was done in DX; there's clamp and clamp-vec.

Reply:
Thats a artifact of  the early days where we wanted to test vector support in `DXILIntrinsicExpansion.cpp` but we were unable to do that type of testing with `DXILLowering.cpp`. Now that we have the scalarizer pass. We should probably clean that up in the DX backend.

Reply:
clamp-vec is now gone. The vector test cases in DirectX clamp.ll now also go through scalarizing and lowering all the way to DXIL ops.